Sources/NostrSDK/Events/ContactListEvent.swift

```diff
@@ -32,6 +32,19 @@ public struct RelayPermissions: Equatable {
 /// > Note: [NIP-02 Specification](https://github.com/nostr-protocol/nips/blob/master/02.md#contact-list-and-petnames)
 public final class ContactListEvent: NostrEvent {
     
+    public required init(from decoder: Decoder) throws {
+        try super.init(from: decoder)
+    }
+    
+    @available(*, unavailable, message: "This initializer is unavailable for this class.")
+    override init(kind: EventKind, content: String, tags: [Tag] = [], createdAt: Int64 = Int64(Date.now.timeIntervalSince1970), signedBy keypair: Keypair) throws {
```

Comment:
I'm trying something fairly opinionated here, so please let me know your feedback. I'm marking the parent class's initializer as unavailable for this class. That way I can prevent client apps from creating an event like this, where the class and the kind don't match:
```swift
let badEvent = ContactListEvent(kind: .recommendServer, tags:...
```

Another great thing about this is that code completion won't even suggest that initializer.

Reply:
Yeah, I think that's a good idea. You can always use the `NostrEvent` initializer if needed for something custom.